### PR TITLE
BP-34: Cluster metadata checker

### PR DESCRIPTION
**Pull request by reddycharan**
Descriptions of the changes in this PR:

- BP for cluster metadata checker

Master Issue: #1602

### CODE REVIEW

**Comment by reddycharan** [comment]
@sijie @eolivelli @jvrao created BP as I presented in the last community meeting.

**Comment by sijie** [comment]
@yzang this metadata cluster checker would be similar as what you have done at Twitter for the visibility on ledger replicas. if you can spend some time reviewing this, that would be great!

File: site/bps/BP-34-cluster-metadata-checker.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+```
+message GetListOfEntriesOfALedgerRequest {
+	required int64 ledgerId = 1;
+}
+
+message GetListOfEntriesOfALedgerResponse {
+	required StatusCode status = 1;
+	required int64 ledgerId = 2;
+	optional bytes availabilityOfEntriesOfLedger = 3; // treated as array of bits indicating availability of entry at that particular index location
```
[the remark this answers is not in the document]
Comment (eolivelli) [reply site/bps/BP-34-cluster-metadata-checker.md]: I don't know if it worth to have a version for a single RPC, wouldn't it be better to have a protocol version ? We don't have versions for each RPC.

The field is "optional" so if we introduce a new "version" we will leave this field *null* and introduce a new one. This is the current approach.

[the remark this answers is not in the document]
Comment (eolivelli) [reply site/bps/BP-34-cluster-metadata-checker.md]: @sijie
I agree, optional fields aren't meant for versioning. My idea wasn't smart

**Comment by sijie** [comment]
run integration tests

File: site/bps/BP-34-cluster-metadata-checker.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (reddycharan) [reply site/bps/BP-34-cluster-metadata-checker.md]: yes, there is validity in having some sort of versioning in this response. Considering we don't have separate version for each request/response, probably it is better to add header/metadata/reserve space at the beginning of 'availabilityOfEntriesOfLedger' bytebuf (byte array). So I suggest to keep 64 bytes reserve space at the beginning, and the first byte is version num, where as the rest is 0s for now.

[the remark this answers is not in the document]
Comment (sijie) [reply site/bps/BP-34-cluster-metadata-checker.md]: +1 lgtm

**Comment by sijie** [comment]
Marked this BP as accepted with 3 approvals (@reddycharan @eolivelli @sijie)

**Comment by sijie** [comment]
IGNORE CI

**Pull request closed without merging**